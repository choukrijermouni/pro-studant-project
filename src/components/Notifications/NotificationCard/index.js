import {
  TypographyElement,
  blue,
  Icon,
  Icons,
  neutral
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import styles from './style'
import Text from './text'
import SmallAvatarWithDot from '../../Common/SmallAvatarWithDot'
import classNames from 'classnames'
import { push } from 'connected-react-router'
import { useDispatch } from 'react-redux'
import { LearnerProfileNeutralPath } from 'Routes'

const useStyle = createUseStyles(styles)

export default ({ unseen, Data = [], ...props }) => {
  const dispatch = useDispatch()
  const { textContainer, icon, avatarContainer, notificationCard } = useStyle({ unseen })
  return (
    <div className={classNames(props.className, notificationCard)} onClick={() => dispatch(push(`${LearnerProfileNeutralPath}/1`))}>
      <div className={avatarContainer}>
        <SmallAvatarWithDot image={Data.image} />
        <div className={textContainer}>
          <div>
            <TypographyElement
              component='p'
              variant='heading4'
              align='left'
              fontSize='16px'
              lignHeight='15px'
              display='inline'
              color={neutral[6]}
            >
              {Data.author || ''}
            </TypographyElement>
            <TypographyElement
              component='p'
              variant='heading4'
              align='left'
              fontSize='16px'
              lignHeight='15px'
              fontWeight='200'
              display='inline'
              color={neutral[4]}
            >
              {Data.action || ''}
            </TypographyElement>
            <TypographyElement
              component='p'
              variant='heading4'
              align='left'
              fontSize='16px'
              lignHeight='15px'
              display='inline'
              color={blue[0]}
            >
              {Data.link || ''}
            </TypographyElement>
          </div>
          <TypographyElement
            component='p'
            variant='heading4'
            align='left'
            fontSize='12px'
            lignHeight='15px'
            color={neutral[4]}
          >
            {Text.time}
          </TypographyElement>
        </div>
      </div>
      <Icon iconName={Icons.roundedUp} style={icon} />
    </div>
  )
}
