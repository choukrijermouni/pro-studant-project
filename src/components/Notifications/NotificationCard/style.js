import {
  neutral,
  blue
} from '@pro_boa/ui'
export default {
  notificationCard: {
    backgroundColor: ({ unseen }) => unseen ? '#F6FCFF' : neutral[0],
    height: 95,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    borderBottom: `.3px ${neutral[2]} solid`,
    padding: 12
  },
  avatarContainer: {
    display: 'flex',
    alignItems: 'start',
    justifyContent: 'flex-start',
    width: '90%'
  },
  icon: {
    color: blue[0],
    fontWeight: 200
  },
  textContainer: {
    marginLeft: 16
  }
}
