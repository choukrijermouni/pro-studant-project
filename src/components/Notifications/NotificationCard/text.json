{
    "name": "Henry Waters",
    "action": " a créé l’équipe",
    "link": " Développement",
    "time": "Il y a 20 min"
}