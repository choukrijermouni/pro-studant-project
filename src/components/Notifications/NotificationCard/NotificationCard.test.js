import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import NotificationCard from '.'
configure({ adapter: new Adapter() })

describe('>>> NotificationCard --- Shallow Render NotificationCard component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<NotificationCard />)
    tree = renderer.create(
      <NotificationCard />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
