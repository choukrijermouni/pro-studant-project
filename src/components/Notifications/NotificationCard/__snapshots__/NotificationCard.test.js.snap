// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> NotificationCard --- Shallow Render NotificationCard component +++ render correctly and match snapshot 1`] = `
<div
  className="notificationCard-0-2-1 notificationCard-d1-0-2-6"
>
  <div
    className="avatarContainer-0-2-2"
  >
    <div
      className="avatarContainer-0-2-8"
    >
      <div
        className="dot-0-2-7"
      />
      <span
        className="avatarGroup-0-2-9 avatarGroup-d0-0-2-10"
      >
        YS
      </span>
    </div>
    <div
      className="textContainer-0-2-4"
    >
      <div>
        <p
          className="text-0-2-11 text-d0-0-2-12"
        >
          
        </p>
        <p
          className="text-0-2-13 text-d0-0-2-14"
        >
          
        </p>
        <p
          className="text-0-2-15 text-d0-0-2-16"
        >
          
        </p>
      </div>
      <p
        className="text-0-2-17 text-d0-0-2-18"
      >
        Il y a 20 min
      </p>
    </div>
  </div>
  <span
    className="icon-roundedUp icon-0-2-3"
  />
</div>
`;
